src/Domain/Creators/BinaryCompactObject.cpp

```diff
@@ -0,0 +1,273 @@
+      outer_radius_objectA_(std::move(outer_radius_objectA)),          // NOLINT
+      xcoord_objectA_(std::move(xcoord_objectA)),                      // NOLINT
+      excise_interiorA_(std::move(excise_interiorA)),                  // NOLINT
+      inner_radius_objectB_(std::move(inner_radius_objectB)),          // NOLINT
+      outer_radius_objectB_(std::move(outer_radius_objectB)),          // NOLINT
+      xcoord_objectB_(std::move(xcoord_objectB)),                      // NOLINT
+      excise_interiorB_(std::move(excise_interiorB)),                  // NOLINT
+      radius_enveloping_cube_(std::move(radius_enveloping_cube)),      // NOLINT
+      radius_enveloping_sphere_(std::move(radius_enveloping_sphere)),  // NOLINT
+      initial_refinement_(                                             // NOLINT
+          std::move(initial_refinement)),                              // NOLINT
+      initial_number_of_grid_points_(                                  // NOLINT
+          std::move(initial_number_of_grid_points)),                   // NOLINT
+      use_equiangular_map_(std::move(use_equiangular_map)),            // NOLINT
+      origin_preimage_(std::move(origin_preimage)),                    // NOLINT
+      use_projective_map_(std::move(use_projective_map))               // NOLINT
+{
+  // Determination of parameters for domain construction:
+  translation_ = 0.5 * (xcoord_objectB_ + xcoord_objectA_);
+  length_inner_cube_ = abs(xcoord_objectA_ - xcoord_objectB_);
+  length_outer_cube_ = 2.0 * radius_enveloping_cube_ / sqrt(3.0);
+  if (length_outer_cube_ <= 2.0 * length_inner_cube_) {
+    double suggested_value = 2.0 * length_inner_cube_ * sqrt(3.0);
+    PARSE_ERROR(
+        context,
+        "The radius for the enveloping cube is too small! The Frustums will be "
+        "malformed. A recommended radius is:\n"
+            << suggested_value);
+  }
+  if (outer_radius_objectA < inner_radius_objectA) {
+    PARSE_ERROR(context,
+                "ObjectA's inner radius must be less than its outer radius.");
+  }
+  if (outer_radius_objectB < inner_radius_objectB) {
+    PARSE_ERROR(context,
+                "ObjectB's inner radius must be less than its outer radius.");
+  }
+  if (use_projective_map_) {
+    projective_scale_factor_ = length_inner_cube_ / length_outer_cube_;
+  } else {
+    projective_scale_factor_ = 1.0;
+  }
+}
+
+template <typename TargetFrame>
+Domain<3, TargetFrame> BinaryCompactObject<TargetFrame>::create_domain() const
+    noexcept {
+  // ObjectA/B may be on the left/right depending on the sign of its xcoord:
+  bool include_center_block_left;
+  bool include_center_block_right;
+  double radius_inner_left;
+  double radius_inner_right;
+  double radius_outer_left;
+  double radius_outer_right;
+  double xcoord_object_left;
+  double xcoord_object_right;
+  double inner_sphericity_left;
+  double inner_sphericity_right;
+
+  if (xcoord_objectB_ > xcoord_objectA_) {
```

Comment:
Could we not just enforce that the inputs put the x-coordinate of B greater than the x coordinate of A, and remove this logic?

Reply:
I will change this to force the x-coordinate of A to be greater than that of B.